src/utils/dump-context.js

```diff
@@ -32,20 +47,78 @@ function tstamp() {
   return retstr;
 }
 
-async function dump(context, action, index) {
-  const nowStr = tstamp();
-  action.debug = action.debug || {};
-  if (!action.debug.dumpDir) {
-    const id = (context.request && context.request.headers && context.request.headers['x-openwhisk-activation-id']) || '';
-    const pathStr = context.request && context.request.path ? context.request.path.replace(/\//g, '-').substring(1) : '';
-    const dirName = `context_dump_${nowStr}_${pathStr}_${id}`;
-    action.debug.dumpDir = path.resolve(process.cwd(), 'logs', 'debug', dirName);
-    await fs.ensureDir(action.debug.dumpDir);
+/**
+ * Ensures that the dump directory exists.
+ * @param {*} context Pipeline context
+ * @param {*} action Pipeline action
+ */
+async function ensureDumpDir(context, action) {
+  if (action.debug.dumpDir) {
+    return;
+  }
+  const stamp = tstamp(new Date());
+  const id = (context.request && context.request.headers && context.request.headers['x-openwhisk-activation-id']) || '';
+  const pathStr = context.request && context.request.path ? context.request.path.replace(/\//g, '-').substring(1) : '';
+  const dirName = `context_dump_${stamp}_${pathStr}_${id}`;
+  action.debug.dumpDir = path.resolve(process.cwd(), 'logs', 'debug', dirName);
+  await fs.ensureDir(action.debug.dumpDir);
+  action.logger.info(`Writing context dumps to: ${action.debug.dumpDir}`);
```

Comment:
should be `debug` level

Reply:
no. in case it dumps because of an error, do I want to see that information in the log right away.